decode/data.ts

```diff
@@ -1,23 +1,76 @@
-import { Event, IDecodedEvent, IPageData, IPingData, ITagData, Token } from "../types/data";
+import { Envelope, Event, MetricData, PageData, PingData, State, SummaryData, TagData, Token, Upload } from "../types/data";
+import { DataEvent, SummaryEvent } from "../types/decode";
 
-export default function(tokens: Token[]): IDecodedEvent {
+let summaries: { [key: number]: SummaryData[] } = null;
+const SUMMARY_THRESHOLD = 250;
+
+export function reset(): void {
+    summaries = {};
+}
+
+export function decode(tokens: Token[]): DataEvent {
     let time = tokens[0] as number;
     let event = tokens[1] as Event;
     switch (event) {
         case Event.Page:
-            let page: IPageData = {
+            let page: PageData = {
                 timestamp: tokens[2] as number,
                 elapsed: tokens[3] as number,
                 url: tokens[4] as string,
-                title: tokens[5] as string,
-                referrer: tokens[6] as string
+                referrer: tokens[5] as string,
+                lean: tokens[6] as number,
```

Comment:
>lean [](start = 16, length = 4)

Is this a boolean? If so can you type it as something that indicates that is the case -- an enum with values for True and False (or On/Off or whatever you think makes sense).

Reply:
Yes good point - using enum now.